index.js

```diff
@@ -119,14 +111,46 @@
   //  $$type :: String
   var $$type = '@@type';
 
-  //  type :: Any -> String
+  //  RPARSE :: RegExp
+  var RPARSE = /^(?:([^/\n\r]+)\/)?(.+?)(?:@(\d+))?$/;
+
+  //       TypeIdentifier :: (a, b, c) -> TypeIdentifier a b c
+  function TypeIdentifier(namespace, name, version) {
+    return {
+      namespace: namespace,
+      name: name,
+      version: version
+    };
+  }
+
+  //       parseTypeIdentifier :: String
+  //                           -> TypeIdentifier String? String Number?
+  function parseTypeIdentifier(s) {
+    var parsed = RPARSE.exec(s);
+    // parsed is null if the string was empty, or contained newlines
```

Comment:
- `'/Type'` is actually parsed as `('', 'Type', null)`, which seems fine to me.†

  - `'@me'` parsing as `(null, '', 'me')` is fine as far as I'm concerned. It seems reasonable to require a version to disambiguate in cases in which the type name contains `@`. Another option is to require the `version` portion to match `^\d+$`, which would allow us to parse this type identifier as `(null, '@me', null)`.

  - `'///'` parsing as `('', '//', null)` seems fine to me.†

  - I agree that the implementation I proposed does not handle `'a/b@2@1'` correctly. Your `lastIndexOf` fix is elegant.

† If you'd like to require the `namespace` to be at least one character in length we could simply begin looking from index `1` rather than from index `0`, but then `'///'` would parse as `('/', '/', null)` which I consider incorrect.